crates/noirc_evaluator/src/ssa_refactor/ir/function_inserter.rs

```diff
@@ -97,6 +97,11 @@ impl<'f> FunctionInserter<'f> {
             InsertInstructionResult::SimplifiedTo(new_result) => {
                 values.insert(old_results[0], *new_result);
             }
+            InsertInstructionResult::SimplifiedToMultiple(new_results) => {
+                for (old_result, new_result) in old_results.iter().zip(new_results.clone()) {
```

Comment:
Why are we cloning new_results here?

Reply:
Removed and I now dereference `new_result` instead